internal/hcs/process.go

```diff
@@ -361,6 +361,52 @@ func (process *Process) CloseStdin(ctx context.Context) error {
 	return nil
 }
 
+func (process *Process) CloseStdout(ctx context.Context) (err error) {
+	ctx, span := trace.StartSpan(ctx, "hcs::Process::CloseStdout") //nolint:ineffassign,staticcheck
+	defer span.End()
+	defer func() { oc.SetSpanStatus(span, err) }()
+	span.AddAttributes(
+		trace.StringAttribute("cid", process.SystemID()),
+		trace.Int64Attribute("pid", int64(process.processID)))
+
+	process.handleLock.Lock()
+	defer process.handleLock.Unlock()
+
+	if process.handle == 0 {
+		return nil
+	}
+
+	process.stdioLock.Lock()
+	defer process.stdioLock.Unlock()
+	if c := process.stdout; c != nil {
```

Comment:
Any reason for the assignment to c instead of just checking `if process.stdout != nil`? I also noticed we explicitly nil out `process.stdin` in `CloseStdin`, do we want to do the same here?

Reply:
I think the assignment to `c` was a vestige of something else, I've removed it now. Some of the specifics of what the contract of the various `Close*` functions should be are unclear, but seems reasonable to assign `nil` after closing given we do the same in `CloseStdin` here.